prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -365,11 +363,10 @@ void efa_rdm_txe_set_runt_size(struct efa_rdm_ep *ep, struct efa_rdm_ope *txe)
 
 	peer = efa_rdm_ep_get_peer(ep, txe->addr);
 
-	iface = txe->desc[0] ? ((struct efa_mr*) txe->desc[0])->peer.iface : FI_HMEM_SYSTEM;
-	hmem_info = &efa_rdm_ep_domain(ep)->hmem_info[iface];
-
 	assert(peer);
-	txe->bytes_runt = MIN(hmem_info->runt_size - peer->num_runt_bytes_in_flight, txe->total_len);
+	txe->bytes_runt = efa_rdm_peer_get_runt_size(peer, ep, txe);
+
+	assert(txe->bytes_runt);
```

Comment:
You have an assert on `txe->bytes_runt` but the value could be zero

Reply:
Oh never mind. If `efa_rdm_peer_get_runt_size` returns 0, then we go to long read. There cannot be a race condition because of the srx lock.